usl_pipeline/usl_lib/usl_lib/readers/polygon_readers.py

```diff
@@ -0,0 +1,47 @@
+import typing
+from typing import Generator, Tuple
+
+from shapely import geometry
+
+
+def read_polygons_from_text_file(
+    file: typing.TextIO,
+) -> Generator[Tuple[geometry.Polygon, int], None, None]:
```

Comment:
We can use a simpler / more general type here like `Iterable[Tuple[geometry.Polygon, int]]`
since we aren't doing fancy stuff with generators like sending values back into the generator.

 I think the iterable type is a little more readable, but either way is fine.

from the docs:
https://docs.python.org/3.6/library/typing.html#typing.Generator
"Alternatively, annotate your generator as having a return type of either Iterable[YieldType] or Iterator[YieldType]:"